From 3da370adf9ebdf9454d797ba16fcdab6acd174db Mon Sep 17 00:00:00 2001
From: Ian Chang <ianchang@ieiworld.com>
Date: Tue, 23 Nov 2021 20:10:55 +0800
Subject: [PATCH 2/2]     1. Add rc.local to turn on LED,Buzzer and run initail
 script     2. Add LAN Port rename function [Puzzle-M902]     3. When OpenVPN
 is running tun on frond end LED0     4. When WAN Port connected turn on frond
 end LED2     5. Add reset to default function when press reset button more
 than 10 Seconds     6. Add uboot partitions define file for firmware and env
 upgrade function

Signed-off-by: Ian Chang <ianchang@ieiworld.com>
---
 .../linux/mvebu/base-files/etc/fw_env.config  | 19 +++++++++++
 .../base-files/etc/openvpn_status.sh          | 18 ++++++++++
 .../mvebu/cortexa72/base-files/etc/rc.local   | 30 +++++++++++++++++
 .../mvebu/cortexa72/base-files/etc/rename.sh  | 33 +++++++++++++++++++
 .../cortexa72/base-files/etc/resetkey.sh      | 33 +++++++++++++++++++
 .../cortexa72/base-files/etc/wan_status.sh    | 18 ++++++++++
 6 files changed, 151 insertions(+)
 create mode 100644 target/linux/mvebu/base-files/etc/fw_env.config
 create mode 100644 target/linux/mvebu/cortexa72/base-files/etc/openvpn_status.sh
 create mode 100644 target/linux/mvebu/cortexa72/base-files/etc/rc.local
 create mode 100755 target/linux/mvebu/cortexa72/base-files/etc/rename.sh
 create mode 100755 target/linux/mvebu/cortexa72/base-files/etc/resetkey.sh
 create mode 100644 target/linux/mvebu/cortexa72/base-files/etc/wan_status.sh

diff --git a/target/linux/mvebu/base-files/etc/fw_env.config b/target/linux/mvebu/base-files/etc/fw_env.config
new file mode 100644
index 0000000000..e4bbcedb62
--- /dev/null
+++ b/target/linux/mvebu/base-files/etc/fw_env.config
@@ -0,0 +1,19 @@
+# Up to two entries are valid, in this case the redundant
+# environment sector is assumed present.
+# Notice, that the "Number of sectors" is ignored on NOR and SPI-dataflash.
+# Futhermore, if the Flash sector size is ommitted, this value is assumed to
+# be the same as the Environment size, which is valid for NOR and SPI-dataflash
+
+# NOR example
+# MTD device name       Device offset   Env. size       Flash sector size       Number of sectors
+#/dev/mtd1              0x0000          0x4000          0x4000
+#/dev/mtd2              0x0000          0x4000          0x4000
+
+# MTD SPI-dataflash example
+# MTD device name       Device offset   Env. size       Flash sector size       Number of sectors
+#/dev/mtd5              0x4200          0x4200
+#/dev/mtd6              0x4200          0x4200
+
+# NAND example
+# device name       Device offset   Env. size       Flash sector size       Number of sectors
+/dev/mtd3               0x0          0x10000         0x10000         1
diff --git a/target/linux/mvebu/cortexa72/base-files/etc/openvpn_status.sh b/target/linux/mvebu/cortexa72/base-files/etc/openvpn_status.sh
new file mode 100644
index 0000000000..877ae30cdf
--- /dev/null
+++ b/target/linux/mvebu/cortexa72/base-files/etc/openvpn_status.sh
@@ -0,0 +1,18 @@
+runFlag=true;
+service_status="running"
+while [[ "$runFlag" = true ]]; do
+  #statements
+  ans=`/etc/init.d/openvpn status`
+  #echo $ans", "${#ans}
+  result=$(echo $ans | grep "${service_status}")
+  if [[ "$result" != "" ]]
+    then
+    #echo "yes"
+    echo -n -e "\x40\x52\x30\x30\x12" > /dev/ttyS1
+  else
+    #echo "no"
+    echo -n -e "\x40\x52\x30\x31\x13" > /dev/ttyS1
+  fi
+  usleep 1000000
+done
+
diff --git a/target/linux/mvebu/cortexa72/base-files/etc/rc.local b/target/linux/mvebu/cortexa72/base-files/etc/rc.local
new file mode 100644
index 0000000000..97382a8d4e
--- /dev/null
+++ b/target/linux/mvebu/cortexa72/base-files/etc/rc.local
@@ -0,0 +1,30 @@
+# Put your custom commands here that should be executed once
+# the system init finished. By default this file does nothing.
+# Set baud rate to 115200
+BOARD=`cat /sys/firmware/devicetree/base/model`
+
+if [ "$BOARD" = "iEi Puzzle-M902" ]
+then
+bash /etc/rename.sh
+fi
+
+if [ "$BOARD" = "iEi Puzzle-M902" ] || [ "$BOARD" = "iEi Puzzle-M901" ]
+then
+stty -F /dev/ttyS1 115200
+# LED4 ON
+echo -n -e "\x40\x52\x33\x30\x11" > /dev/ttyS1
+# Fan full speed
+echo -n -e "\x40\x46\x57\x30\xFF\x9E" > /dev/ttyS1
+# BUZZER 1.5s
+echo -n -e "\x40\x43\x33\x30" > /dev/ttyS1
+
+sleep 1
+# LED3 ON
+echo -n -e "\x40\x52\x32\x30\x10" > /dev/ttyS1
+
+bash /etc/openvpn_status.sh &
+bash /etc/wan_status.sh &
+bash /etc/resetkey.sh &
+fi
+
+exit 0
diff --git a/target/linux/mvebu/cortexa72/base-files/etc/rename.sh b/target/linux/mvebu/cortexa72/base-files/etc/rename.sh
new file mode 100755
index 0000000000..813ee1a9d0
--- /dev/null
+++ b/target/linux/mvebu/cortexa72/base-files/etc/rename.sh
@@ -0,0 +1,33 @@
+#!/bin/bash
+BOARD=`cat /sys/firmware/devicetree/base/model`
+
+if [ "$BOARD" = "iEi Puzzle-M902" ] 
+then
+/etc/init.d/network stop
+nicslist="eth0 eth1 eth2 eth3 eth4 eth5 eth6 eth7 eth8"
+      for nics in $nicslist
+         do  
+         ifconfig $nics down
+         echo down_nic=$nics
+      done
+#ifconfig -a | grep HW | awk '{ print $1" : "$5 }' > ${testfile}
+      for nics in $nicslist
+         do
+         IFACE=$nics
+         read MAC </sys/class/net/$IFACE/address
+         #echo $IFACE $MAC
+           case $nics in   
+             eth0) nameif eth10 $MAC;; ##10G
+             eth1) echo "eth1" $MAC;;
+             eth2) nameif eth0 $MAC;;
+             eth3) nameif eth11 $MAC;; ##10G
+             eth4) nameif eth3 $MAC;;
+             eth5) nameif eth2 $MAC;;
+             eth6) nameif eth12 $MAC;; ##10G
+             eth7) nameif eth5 $MAC;;
+             eth8) nameif eth4 $MAC;;
+           esac
+       done
+sleep 1
+/etc/init.d/network restart
+fi
diff --git a/target/linux/mvebu/cortexa72/base-files/etc/resetkey.sh b/target/linux/mvebu/cortexa72/base-files/etc/resetkey.sh
new file mode 100755
index 0000000000..2588a9b46b
--- /dev/null
+++ b/target/linux/mvebu/cortexa72/base-files/etc/resetkey.sh
@@ -0,0 +1,33 @@
+runFlag=true;
+checkCount=0;
+maxCheckCount=20;
+while [[ "$runFlag" = true ]]; do
+  if (($checkCount >= $maxCheckCount));then
+      # echo "pass to run";
+      checkCount=0
+      sysupgrade -r /etc/backup-puzzle-m90x.tar.gz
+      # sysupgrade -r /etc/back-fake-.tar.gz
+      # sysupgrade -r /etc/back-fake-.tar.g
+      if [ $? == 0 ]; then
+	echo -n -e "\x40\x52\x32\x32\x12" > /dev/ttyS1 
+        echo -n -e "\x40\x43\x32\x31" > /dev/ttyS1   
+        runFlag=false;	
+      fi
+  fi
+  #statements
+  ans=`devmem 0xf2440150`
+  # echo $ans", "${#ans}
+  if ((${#ans} == 10));then
+      ans=${ans: 8: 1}
+      ans=$((16#${ans}%2))
+      if [[ $ans = 1 ]]; then
+      checkCount=0;
+      else
+      # press reset button
+      checkCount=$(( $checkCount + 1 ))
+      fi
+  fi
+  usleep 500000
+done
+
+reboot
diff --git a/target/linux/mvebu/cortexa72/base-files/etc/wan_status.sh b/target/linux/mvebu/cortexa72/base-files/etc/wan_status.sh
new file mode 100644
index 0000000000..c7a26403e9
--- /dev/null
+++ b/target/linux/mvebu/cortexa72/base-files/etc/wan_status.sh
@@ -0,0 +1,18 @@
+runFlag=true;
+link_yes="yes"
+link_no="no"
+while [[ "$runFlag" = true ]]; do
+  #statements
+  ans=`ethtool eth0 | grep -i 'Link det'`
+  #echo $ans", "${#ans}
+  result=$(echo $ans | grep "${link_yes}")
+  if [[ "$result" != "" ]]
+    then
+    #echo "yes"
+    echo -n -e "\x40\x52\x31\x30\x13" > /dev/ttyS1
+  else
+    #echo "no"
+    echo -n -e "\x40\x52\x31\x31\x12" > /dev/ttyS1
+  fi
+  usleep 1000000
+done
-- 
2.17.1

